game/forms.py

```
@@ -1,8 +1,32 @@
+
+    def clean_username(self):
+        username = self.cleaned_data.get('username', '')
+
+        if username != username.strip():
+            raise forms.ValidationError(
+                'Username must not have leading or trailing spaces.'
+            )
+
+        if not USERNAME_RE.match(username):
+            raise forms.ValidationError(
+                'Username must be 3–20 characters and may only contain '
```

Ruff (0.15.12)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)